kafkastats/src/main/java/com/pinterest/doctorkafka/stats/KafkaAvroPublisher.java

```diff
@@ -78,8 +81,16 @@ public void publish(BrokerStats brokerStats) throws IOException {
       int numPartitions = kafkaProducer.partitionsFor(destTopic).size();
       int partition = brokerStats.getId() % numPartitions;
 
-      Future<RecordMetadata> future = kafkaProducer.send(
-          new ProducerRecord<>(destTopic, partition, key.getBytes(), stream.toByteArray()));
+      ProducerRecord<byte[], byte[]> record = null;
+
+      if (legacyMode) {
```

Comment:
do we need this change? 

https://kafka.apache.org/21/javadoc/org/apache/kafka/clients/producer/ProducerRecord.html

    The record also has an associated timestamp. If the user did not provide a timestamp, the producer 
    will stamp the record with its current time. The timestamp eventually used by Kafka depends on 
    the timestamp type configured for the topic.

    If the topic is configured to use CreateTime, the timestamp in the producer record will be 
    used by the broker.  If the topic is configured to use LogAppendTime, the timestamp in 
    the producer record will be overwritten by the broker with the broker local time when it 
     appends the message to its log.

Reply:
Sorry I forgot about this; let me open a PR for the server side.

Reply:
Created a new PR with the serverside changes
https://github.com/pinterest/doctorkafka/pull/64